app/recorder.js

```diff
@@ -1,12 +1,24 @@
-module.exports = app => (name, options) => {
+const UrlPattern = require('url-pattern');
+
+module.exports = app => (name, options = {}) => {
+  const { match } = options;
+  let matchFn = () => true;
+
   app.locals.recording = true;
   if (!name) throw new Error('Must provide a recording name.');
 
   app.log(['serve', 'record'], name);
 
+  if (match) {
+    const urlPattern = new UrlPattern(match);
+    app.log(['serve', 'record', 'match'], urlPattern.stringify());
+    matchFn = urlPattern.match.bind(urlPattern);
+  }
+
   app.locals.recordMeta = {
     name,
-    options
+    options,
+    match: matchFn
```

Comment:
So `match` is already in `options`, right? Why not just clone that and update `match` to be the compiled `matchFn`? Then in the future we could also support a programmatic API where user is allowed to pass `match` option as a custom function we will call (allowing glob or other rules that oculd potentially match on the whole request and/or response).

Reply:
Is this what you are suggesting?
```js
  let { match } = options;

  app.locals.recording = true;
  if (!name) throw new Error('Must provide a recording name.');

  app.log(['serve', 'record'], name);

  if (typeof match === 'string') {
    // if match is a string, assume it is a url pattern
    const urlPattern = new UrlPattern(match);
    app.log(['serve', 'record', 'match'], urlPattern.stringify());
    match = urlPattern.match.bind(urlPattern);
  }

  if (!match) {
    // if match is not defined, assign noop which matches all requests
    match = () => true;
  }

  app.locals.recordMeta = {
    name,
    options,
    match
  };
```

Reply:
Yes, except I think you could keep `match` in `options`. And we don't necessarily need the truthy noop, we can handle no match option later on.

Reply:
True, on the noop... on second thought, the responsibility does seem well placed here.